src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
+        private static readonly OSName s_linux = new OSName(LinuxName);
+        private static readonly OSName s_osx = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name.Equals("unix", StringComparison.OrdinalIgnoreCase))
```

Comment:
Directly using `string` has a number of disadvantages:

- A simple `string` lacks the context to find the set of related valid identifiers.  If I see a parameter `string osname` how do I navigate to the set of valid values in that domain?  With the struct `OSName` navigating to the set of valid values is trivial.
- All `string` values have to answer the question of which `StringComparer` instance to use?  Digging through the docs to find the right value is wasted developer time.  With `OSName` there is no need to research, just use `==` and move on.